extensions/amp-subscriptions-google/0.1/amp-subscriptions-google.js

```diff
@@ -496,53 +506,86 @@ export class GoogleSubscriptionsPlatform {
-          // by a hash value which would allow non gooogle hostnames to construct LAA urls.
-          ((parsedReferrer.protocol === 'https:' &&
-            GOOGLE_DOMAIN_RE.test(parsedReferrer.hostname)) ||
-            getMode(this.ampdoc_.win).localDev) &&
-          parsedQuery[`gaa_at`] == 'la' &&
-          parsedQuery[`gaa_n`] &&
-          parsedQuery[`gaa_sig`] &&
-          parsedQuery[`gaa_ts`] &&
-          parseInt(parsedQuery[`gaa_ts`], 16) > Date.now() / 1000
-        ) {
-          // All the criteria are met to return an LAA entitlement
-          return Promise.resolve(
-            new Entitlement({
-              source: 'google:laa',
-              raw: '',
-              service: PLATFORM_ID,
-              granted: true,
-              grantReason: GrantReason.LAA,
-              dataObject: {},
-              decryptedDocumentKey: null,
-            })
-          );
-        }
+    return this.getGoogleMeteringStrategy_().then((strategy) => {
+      // Verify Google's metering strategy for this article.
+      if (strategy !== GoogleMeteringStrategy.LEAD_ARTICLE) {
+        return null;
+      }
+
+      // All the criteria are met to return an LAA entitlement
+      return new Entitlement({
+        source: 'google:laa',
+        raw: '',
+        service: PLATFORM_KEY,
+        granted: true,
+        grantReason: GrantReason.LAA,
+        dataObject: {},
+        decryptedDocumentKey: null,
       });
+    });
+  }
+
+  /**
+   * Returns Google's metering strategy for this article.
+   * @private
+   * @return {!Promise<!GoogleMeteringStrategy>}
+   */
+  getGoogleMeteringStrategy_() {
+    // Verify the service config enables a Google metering strategy.
+    if (!this.enableLAA_ && !this.enableMetering_) {
+      return Promise.resolve(GoogleMeteringStrategy.NONE);
     }
-    return Promise.resolve(null);
+
+    return this.viewerPromise_.getReferrerUrl().then((referrer) => {
+      // Check referrer.
+      const parsedReferrer = parseUrlDeprecated(referrer);
+      if (
+        (parsedReferrer.protocol !== 'https:' ||
+          !GOOGLE_DOMAIN_RE.test(parsedReferrer.hostname)) &&
+        // Note we don't use the more generic this.isDev_ flag because that can be
+        // triggered by a hash value which would allow non google hostnames to
+        // construct LAA urls.
+        !getMode(this.ampdoc_.win).localDev
+      ) {
+        return GoogleMeteringStrategy.NONE;
+      }
+
+      // Parse URL params.
+      const urlParams = this.getUrlParams_();
+
+      // Verify timestamp.
+      if (parseInt(urlParams[`gaa_ts`], 16) < Date.now() / 1000) {
+        return GoogleMeteringStrategy.NONE;
+      }
+
+      // Verify a few params exist.
+      if (!urlParams[`gaa_n`] || !urlParams[`gaa_sig`]) {
+        return GoogleMeteringStrategy.NONE;
+      }
+
+      // Determine Google metering strategy.
+      if (urlParams[`gaa_at`] === 'la') {
```

Comment:
I think we should be using `if (this.enableLAA_)` here. It's reasonable that a user would see extended access when visiting from a lead article.

Reply:
Good catch! Updated